### PR TITLE
Add support for argocd and custom annotations to the init job

### PR DESCRIPTION
**Pull request by codelite7**
Added argocd sync hook to the init job annotations.
Added additional annotations configuration fo the init job.
This addresses https://github.com/cockroachdb/helm-charts/issues/149

### CODE REVIEW

**Comment by diogokiss** [comment]
Can this PR get some love soon? ❤️  I'm also affected by this issue. 😢

**Comment by rail** [comment]
@pseudomuto maybe you can review this?

File: cockroachdb/templates/job.init.yaml [hunk 1]
```diff
@@ -21,6 +21,10 @@ metadata:
   annotations:
     helm.sh/hook: post-install,post-upgrade
     helm.sh/hook-delete-policy: before-hook-creation
+    argocd.argoproj.io/hook: Sync
```
[the remark this answers is not in the document]
Comment (diogokiss) [reply cockroachdb/templates/job.init.yaml:24]: I thought the same, @pseudomuto.

**Comment by diogokiss** [comment]
> I think we should add/update a test here just to ensure this is exercised and not lost in a future update.

@pseudomuto, what kind of test exactly would you like to be implemented?

**Comment by diogokiss** [comment]
@codelite7, please, let me know if you would like me to contribute. Considering the Github profile, I'd assume that @pseudomuto is based in Canada. But I don't know where you're based in. So, as I'm in Europe (CEST) there might be some time zone difference that would delay things a bit, unfortunately.

**Comment by diogokiss** [comment]
@pseudomuto, @rail, and @codelite7, I've ported the changes from this PR to https://github.com/cockroachdb/helm-charts/pull/250 to include the changes suggested throughout this thread. Please, give it some love. ❤️

**Pull request closed without merging**